### ISSUE TITLE
supportedRuntime arguments lost in merging

### ISSUE THREAD

**Issue by borisdan** [Microsoft/ILMerge#59]
Hello,
I have a project which targets .NET 3.5, but should run successfully on an environment where only .NET 4.0 is installed. Here is my app.config:

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
    <startup useLegacyV2RuntimeActivationPolicy="true">
        <supportedRuntime version="v4.0" />
        <supportedRuntime version="v2.0.50727"/>
    </startup>
</configuration>
```

As per [Microsoft's documentation](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-configure-an-app-to-support-net-framework-4-or-4-5).

I run ILMerge with the targetPlatform flag:
> /targetplatform:v2,C:/Windows/Microsoft.NET/Framework64/v2.0.50727

However, this path doesn't exist on a system without .NET3.5. When I run the merged exe, it fails, with a window notifying on a failure to install .NET3.5.
However, when I copy the entire built folder of DLLs and the original EXE and run it, it succeeds. The original EXE is compiled to .NET3.5, and runs successfully on a machine without .NET 4.0 as well.

I was looking for a way to use .NET's supportedRuntime configuration in the merged EXE, but I can't find one. Is there a workaround? Or is it just a missing feature?

**Comment by mike-barnett** [comment]
Sorry, but I'm afraid I don't completely understand the problem. The /targetplatform option for ILMerge should be needed only when the .NET runtime you are executing ILMerge in is different from the one that you want the merged assembly to execute in. The path is just an example: if you need the option then you need to replace it with the path that points to the .NET installation that you want the merged assembly to execute in.  It may be that you just need to create a file called "foo.exe.config" where "foo" is the name of the merged assembly. That is what happens to the app.config file when you compile an executable. It just gets renamed and copied into the output directory where the .exe lives.

Please let me know if any of this helps. Thanks!

**Comment by borisdan** [comment]
Hi @mike-barnett,
It is my fault, I misunderstood. In fact, I didn't realize just how complicated my problem is.
I use a framework which allows me to attach one specific EXE to an MSI, in order to do some task. The MSI needs to run on environments that have .NET 3.5, .NET 4.0, or both. I wanted to leverage the supportedRuntime configs in the App.config file, but I realized it only works when the file is outside the EXE: It can't be merged with ILMerge, and if it is marked as embedded on the project, then it can only be loaded after the appropriate runtime has been chosen. The next possibility was to write an unmanaged loader, but then ILMerge can't merge it again.
I ended up rewriting the whole thing in C++, as the VC Redistributable is the only thing that is guaranteed to be on the target machine, because I bundle it in with the MSI.
Boris
